gsa/src/web/authorized.js

```diff
@@ -15,108 +15,86 @@
 
-import {withRouter} from 'react-router-dom';
+import {useDispatch} from 'react-redux';
 
-import {isDefined} from 'gmp/utils/identity';
+import {useQuery} from '@apollo/react-hooks';
 
-import {setIsLoggedIn} from './store/usersettings/actions';
-import {isLoggedIn} from 'web/store/usersettings/selectors';
+import gql from 'graphql-tag';
 
-import compose from 'web/utils/compose';
-import PropTypes from 'web/utils/proptypes';
-import withGmp from 'web/utils/withGmp';
+import {hasValue} from 'gmp/utils/identity';
 
-class Authorized extends React.Component {
-  constructor(...args) {
-    super(...args);
+import Loading from 'web/components/loading/loading';
 
-    this.responseError = this.responseError.bind(this);
-  }
-
-  componentDidMount() {
-    const {gmp} = this.props;
-
-    this.responseError = this.responseError.bind(this);
+import {setIsLoggedIn} from 'web/store/usersettings/actions';
 
-    this.unsubscribe = gmp.addHttpErrorHandler(this.responseError);
-
-    this.checkIsLoggedIn();
-  }
+import useGmp from 'web/utils/useGmp';
 
-  componentWillUnmount() {
-    if (isDefined(this.unsubscribe)) {
-      this.unsubscribe();
+const GET_CURRENT_USER = gql`
+  query {
+    currentUser {
+      isAuthenticated
     }
   }
+`;
 
-  componentDidUpdate() {
-    this.checkIsLoggedIn();
-  }
-
-  responseError(xhr) {
-    const {logout} = this.props;
-
-    if (xhr.status === 401) {
-      logout();
-      return Promise.resolve(xhr);
-    }
-    return Promise.reject(xhr);
-  }
+const useQueryCurrentUser = () =>
+  useQuery(GET_CURRENT_USER, {
+    fetchPolicy: 'no-cache',
+  });
 
-  checkIsLoggedIn() {
-    if (!this.props.isLoggedIn) {
-      this.toLoginPage();
-    }
-  }
+const Authorized = ({children}) => {
+  const {data, loading} = useQueryCurrentUser();
+  const [isLoading, setIsLoading] = useState(loading);
 
-  toLoginPage() {
-    const {history, location} = this.props;
+  const gmp = useGmp();
+  const location = useLocation();
+  const history = useHistory();
+  const dispatch = useDispatch();
 
+  const toLoginPage = () => {
     if (location.pathname === '/login') {
       // already at login page
       return;
     }
 
     history.replace('/login', {
-      next: this.props.location.pathname,
+      next: location.pathname,
     });
-  }
+  };
+
+  useEffect(() => {
+    const unsubscribe = gmp.subscribeToLogout(() => toLoginPage());
+    return unsubscribe;
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  render() {
-    return this.props.isLoggedIn ? this.props.children : null;
+  useEffect(() => {
+    if (hasValue(data)) {
+      const {isAuthenticated} = data.currentUser;
+
+      dispatch(setIsLoggedIn(isAuthenticated));
+
+      if (!isAuthenticated) {
+        toLoginPage();
+      }
+    }
+    setIsLoading(loading);
+  }, [loading, data]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  if (isLoading) {
+    // to prevent infinite loading if there is an error
```

Comment:
I think these comments are no longer relevant and might be confusing at a later date.